cl/custom_filters/templatetags/extras.py

```diff
@@ -247,6 +267,40 @@ def get_highlight(result: AttrDict | dict[str, any], field: str) -> any:
     return render_string_or_list(hl_value) if hl_value else original_value
 
 
+@register.simple_tag
+def extract_q_value(query: str) -> str:
+    """Extract the value of the "q" parameter from a URL-encoded query string.
+
+    :param query: The URL-encoded query string.
+    :return: The value of the "q" parameter or an empty string if "q" is not found.
+    """
+
+    parsed_query = urllib.parse.parse_qs(query)
+    return parsed_query.get("q", [""])[0]
+
+
+@register.simple_tag(takes_context=True)
+def alerts_supported(context: RequestContext, search_type: str) -> str:
+    """Determine if search alerts are supported based on the search type and flag
+    status.
+
+    :param context: The template context, which includes the request, required
+    for the waffle flag.
+    :param search_type: The type of search being performed.
+    :return: True if alerts are supported, False otherwise.
+    """
+
+    request = context["request"]
+    return (
+        search_type == SEARCH_TYPES.OPINION
+        or search_type == SEARCH_TYPES.ORAL_ARGUMENT
+        or (
+            search_type == SEARCH_TYPES.RECAP
+            and waffle.flag_is_active(request, "recap-alerts-active")
+        )
+    )
```

Comment:
I believe we can refactor this code to make it more readable.

```suggestion
    request = context["request"]
    if search_type == SEARCH_TYPES.RECAP:
        return waffle.flag_is_active(request, "recap-alerts-active")
    return search_type in (SEARCH_TYPES.OPINION, SEARCH_TYPES.ORAL_ARGUMENT)
```

Reply:
yeah, this looks good!